src/compile.coffee

```diff
@@ -52,35 +57,70 @@ compile = (mediaType, apiElements, filename) ->
   {transactions, errors, warnings}
 
 
-findRelevantTransactions = (mediaType, parseResult) ->
+findRelevantTransactions = (mediaType, refract, apiElements) ->
   relevantTransactions = []
 
-  if mediaType is 'text/vnd.apiblueprint'
+  # This gets deleted once we're fully on minim
+  refractTransitions = children(refract, {element: 'transition'})
+
+  apiElements.findRecursive('transition').forEach((transition, transitionNo) ->
+    transitionNo = transitionNo.toValue()
+
+    # This gets deleted once we're fully on minim
+    refractTransition = refractTransitions[transitionNo]
+    refractHttpTransactions = children(refractTransition, {element: 'httpTransaction'})
+
     # API Blueprint has a concept of transaction examples and
     # the API Blueprint AST used to expose it. The concept isn't present
     # in API Elements anymore, so we have to detect and backport them.
-    children(parseResult, {element: 'transition'}).map(detectTransactionExamples)
-
-  # Dredd supports only testing of the first request-response pair within
-  # each transaction example. So if we're dealing with API Blueprint, we
-  # iterate over available transactions and skip those, which are not first
-  # within a particular example.
-  #
-  # That's achieved by tracking example number within each transition
-  # and skipping transactions with example number we've already seen.
-  for transition in children(parseResult, {element: 'transition'})
-    example = 0
+    if mediaType is 'text/vnd.apiblueprint'
```

Comment:
This is very specific to API Blueprint and to backwards compatibility of Dredd. We need the transaction example numbers only because transaction names for hooks contain them. There's a plan to migrate to so-called "transaction paths" in the future (https://github.com/apiaryio/dredd/issues/227), which won't use the concept of a transaction examples anymore. So I'd rather keep this isolated and ready to be removed in the future.